pkg/registration/spoke/spokeagent.go

```diff
@@ -170,10 +170,52 @@ func (o *SpokeAgentConfig) RunSpokeAgentWithSpokeInformers(ctx context.Context,
 	namespacedManagementKubeInformerFactory := informers.NewSharedInformerFactoryWithOptions(
 		managementKubeClient, 10*time.Minute, informers.WithNamespace(o.agentOptions.ComponentNamespace))
 
+	// bootstrapKubeConfig is the file path of the bootstrap kubeconfig. If MultipleHubs feature is enabled, it is
+	// the selected bootstrap kubeconfig file path.
+	//
+	// hubValiedConfig is a function to check if the hub client config is valid.
+	var bootstrapKubeConfig string
+	var hubValidConfig func(ctx context.Context) (bool, error)
+	if features.SpokeMutableFeatureGate.Enabled(ocmfeature.MultipleHubs) {
+		// bootstrapkubeconfig
+		var bootstrapKubeConfigs []*rest.Config
+		for _, fp := range o.registrationOption.BootstrapKubeconfigs {
+			kc, err := clientcmd.BuildConfigFromFlags("", fp)
+			if err != nil {
+				return fmt.Errorf("unable to load bootstrap kubeconfig from file %q: %w", fp, err)
+			}
+			bootstrapKubeConfigs = append(bootstrapKubeConfigs, kc)
+		}
+
+		index, err := selectBootstrapKubeConfigs(ctx, o.agentOptions.SpokeClusterName, bootstrapKubeConfigs)
+		if err != nil {
+			return fmt.Errorf("failed to select a bootstrap kubeconfig: %w", err)
+		}
+		logger.Info("Select bootstrap kubeconfig", "index", index, "file", o.registrationOption.BootstrapKubeconfigs[index])
+
+		bootstrapKubeConfig = o.registrationOption.BootstrapKubeconfigs[index]
+
+		// hubValiedConfig
+		hubValidConfig = func(ctx context.Context) (bool, error) {
+			return o.HasValidHubClientConfig(ctx)
```

Comment:
why we do not compare hub and boostratp config here?

Reply:
because the hub-kubeconfig and the bootstrap-kubeconfig are not always should be equal.

For example, we have a [hub1, hub2],both hub1 and hub2 are available now, currently the agent connect to hub2. So the hub-kubeconfig is point to hub2.

Now we restart the agent, with our select logic, the bootstrapkubeconfig we choose will point to hub1. But we still want to hub-kubeconfig points to hub2.

If we compare the bootstrapkubeconfig and hub-kubeconfig, the agent will fail on this line:
```
        // check if there already exists a valid client config for hub
	ok, err := hubValidConfig(ctx)
	if err != nil {
		return err
	}
```

Reply:
So that is about how we select bootstrap right? I think we can do like this:

1. if there is kubeKubeconfig, get the part we want to compare with bootstrap, and check the matched bootstrap at first.
2 if the matched boostratp is found and check passed. skip the rebootstrap
3. otherwise, find the bootstrap based on order.

Reply:
It's a great point, updated accordingly.